source/extensions/transport_sockets/tls/ssl_handshaker.cc

```diff
@@ -242,7 +242,7 @@ Network::PostIoAction SslHandshakerImpl::doHandshake() {
       return PostIoAction::KeepOpen;
     default:
       handshake_callbacks_->onFailure();
-      return PostIoAction::Close;
+      return PostIoAction::CloseError;
```

Comment:
Does the PostIoAction::Close on line 233 above a graceful or error close?

Reply:
Graceful.